### ISSUE TITLE
GPU & settings to successfully train on image resolution higher than (448, 320, 3)

### ISSUE THREAD

**Issue by Avi-avidan** [matterport/Mask_RCNN#457]
Hi,
This repo is very useful. thanks Matterport and other contributers.
I am getting nice results on low res images (mAP ~ 0.85), but when trying to train on higher resolution, process fails/stuck/not running.

I would love to get some tips on how to train for higher res. and if anyone can share configuration + GPU used to successfully train on high res images.

THANKS A LOT.

This is my HW - 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.582
pciBusID: 0000:01:00.0
totalMemory: 10.92GiB freeMemory: 10.38GiB
tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10044 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)

This is my last config details -

Configurations:
BACKBONE                       resnet50
BACKBONE_SHAPES                [[80 80]
 [40 40]
 [20 20]
 [10 10]
 [ 5  5]]
BACKBONE_STRIDES               [4, 8, 16, 32, 64]
BATCH_SIZE                     1
BBOX_STD_DEV                   [0.1 0.1 0.2 0.2]
DETECTION_MAX_INSTANCES        20
DETECTION_MIN_CONFIDENCE       0.65
DETECTION_NMS_THRESHOLD        0.3
GPU_COUNT                      1
GRADIENT_CLIP_NORM             2
IMAGES_PER_GPU                 1
IMAGE_MAX_DIM                  320
IMAGE_META_SIZE                14
IMAGE_MIN_DIM                  320
IMAGE_MIN_SCALE                0
IMAGE_RESIZE_MODE              pad64
IMAGE_SHAPE                    [320 320   3]
LEARNING_MOMENTUM              0.9
LEARNING_RATE                  0.001
LOSS_WEIGHTS                   {'rpn_class_loss': 1.0, 'rpn_bbox_loss': 1.0, 'mrcnn_class_loss': 1.0, 'mrcnn_bbox_loss': 1.0, 'mrcnn_mask_loss': 1.0}
MASK_POOL_SIZE                 14
MASK_SHAPE                     [28, 28]
MAX_GT_INSTANCES               100
MEAN_PIXEL                     [138.25 128.33 118.01]
MINI_MASK_SHAPE                (56, 56)
NAME                           Post_Zeekit_320_
NUM_CLASSES                    2
POOL_SIZE                      7
POST_NMS_ROIS_INFERENCE        1000
POST_NMS_ROIS_TRAINING         2000
ROI_POSITIVE_RATIO             0.33
RPN_ANCHOR_RATIOS              [0.5, 1, 2]
RPN_ANCHOR_SCALES              (32, 64, 128, 256, 512)
RPN_ANCHOR_STRIDE              1
RPN_BBOX_STD_DEV               [0.1 0.1 0.2 0.2]
RPN_NMS_THRESHOLD              0.7
RPN_TRAIN_ANCHORS_PER_IMAGE    128
STEPS_PER_EPOCH                3200
TRAIN_BN                       False
TRAIN_ROIS_PER_IMAGE           56
USE_MINI_MASK                  True
USE_RPN_ROIS                   True
VALIDATION_STEPS               320
WEIGHT_DECAY                   0.0001

**Comment by waleedka** [comment]
There are a few ideas you can try to reduce memory requirements. Unfortunately, most of them do affect the training and might cause lower accuracy. 

1. Use smaller batch size.
2. Use fewer ROIs for training the second stage (`config.TRAIN_ROIS_PER_IMAGE`)
3. Reduce `config.MAX_GT_INSTANCES` is it's higher than what you need
4. Train fewer layers. Rather than all layers, you can train 3+ or 4+.
5. Train on crops of images rather than the full images. See `config.IMAGE_RESIZE_MODE = 'crop'`




If it hangs, it's probably a memory error. I've noticed that Keras, in some cases, just hangs when TensorFlow fails to allocate memory it needs. If it takes too long then stop it and see if it prints a memory error.

**Comment by Avi-avidan** [comment]
thanks for all the tips. Great repo.

בתאריך יום ה׳, 19 באפר׳ 2018, 00:10, מאת Waleed ‏<notifications@github.com>:

**Comment by waleedka** [comment]
In case others stumble upon this thread, I put a summary of memory optimization tips here: https://github.com/matterport/Mask_RCNN/wiki
